src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -788,6 +766,88 @@ public SolutionCompilationState WithProjectAnalyzerReferences(
                     stateChange.OldProjectState, stateChange.NewProjectState, referencesToAdd: addedReferences, referencesToRemove: removedReferences);
             },
             forkTracker: true);
+
+        // As we're changing the references for a project we want to force generators to rerun for it.  This way the
+        // user automatically sees the new generators values without having to take any more explicit steps.
+        var updatedMap = GetUpdatedSourceGeneratorVersions(this,
```

Comment:
is it possible that there are not actually any changed references here, and we end up force re-generation needlessly?  Or does the caller guarantee that there are definitely changed references?

Reply:
nevermind - seems like the callers are guaranteeing that

Reply:
yup.  callers guarantee